### ISSUE TITLE
Label is parsed incorrectly and gives a parsing error

### ISSUE THREAD

**Issue by jamesWalker55** [JohnnyMorganz/StyLua#309]
I'm using StyLus v0.11.2 on Windows. I have a file that looks like this:

```lua
for i, v in ipairs(things) do
  foo = 1
  ::label::
end
```

Upon running `stylua .\`, it gives the following error:

```
error: could not format file .\lua\test.lua: error parsing: unexpected token `end`. (starting from line 4, character 1 and ending on line 4, character 4)                                                                                                
additional information: expected identifier after `::`
```

I believe the code above follows the label syntax correctly, I'm not sure what's causing this error?

**Comment by JohnnyMorganz** [comment]
How are you downloading stylua? Is it via cargo or through a download?

I think this might be a conflict between the `luau` feature and the `lua52` one, which are both enabled by default in the prebuilt binaries. If you install using `cargo install stylua --features lua52`, does it work as expected?

**Comment by JohnnyMorganz** [comment]
Yeah, the underlying issue is in our [parser library](https://github.com/Kampfkarren/full-moon) when both features are enabled at the same time (Luau has syntax for type assertions after expressions like `local x = 1 :: number`, which conflicted with your label syntax here)

> Would it be possible to provide binaries for both `luau` and `lua52` versions in the releases page? The compilation for the `lua52` took a while on my slow computer, I feel it would be more convenient to provide two binaries directly.

This is definitely something I want to do - #103, the only issue right now is within a toolchain manager that some commonly use alongside stylua (https://github.com/Roblox/foreman). If there are multiple different possible binaries to install using foreman it currently leads to undefined behaviour.
